hkmc2/shared/src/test/mlscript/std/RenderingTest.mls

```diff
@@ -316,3 +330,118 @@ open Stack
 
 renderAsStr of 1 :: 2 :: 3 :: 4
 //│ = "Cons(1, Cons(2, Cons(3, 4)))"
+
+class Origin with
+  val x = 0
+  val y = 0
+
+renderAsStr of new Origin
+//│ = "Origin {x: 0, y: 0}"
+
+class Point(x: Num, y: Num) with
+  val x = x
+  val y = y
+
+// Should we display the members?
+renderAsStr of new Point(0, 1)
+//│ = "Point()"
```

Comment:
Feels like we should render this as
```suggestion
//│ = "Point(_, _)"
```
to match the way this would be matched as a patterm.

(In general, I think the syntax pretty-printed stuff mostly corresponds to the syntax of corresponding patterns.)

Reply:
Hmm if that's really the case, it feels like we might want to display
```fs
//│ = "(val ref'1 = SelfRef {a: ref'1, b: 1})"
```
as
```fs
//│ = "SelfRef {a: ref'1, b: 1} as ref'1"
```
instead, corresponding to a pattern or type (this should be both a valid pattern and a valid type).

This also has the advantage of showing the more important stuff first (the object's constructor).

What do you think?